DocumentDataAPI/DocumentDataAPI/Controllers/DocumentController.cs

```diff
@@ -0,0 +1,208 @@
+﻿using DocumentDataAPI.Data.Repositories;
+using DocumentDataAPI.Models;
+
+using Microsoft.AspNetCore.Mvc;
+
+using System.Net.Mime;
+
+namespace DocumentDataAPI.Controllers
+{
+    [ApiController]
+    [Route("document")]
+    [Produces(MediaTypeNames.Application.Json)]
+    public class DocumentController : ControllerBase
+    {
+        private readonly ILogger<DocumentController> _logger;
+        private readonly IDocumentRepository _repository;
+
+        public DocumentController(ILogger<DocumentController> logger, IDocumentRepository repository)
+        {
+            _logger = logger;
+            _repository = repository;
+        }
+
+        /// <summary>
+        /// Adds the document from the content body.
+        /// </summary>
+        /// <response code="200">Success: The document that was added to the database.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpPut]
+        [Route("PutDocument")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<DocumentModel> PutDocument([FromBody] DocumentModel document)
+        {
+            try
+            {
+                return _repository.Add(document) == 0
+                    ? Problem("No rows were added")
+                    : Ok(_repository.Get(document.Id));
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to add document.\n{e.Message}");
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a list of all documents from the database.
+        /// </summary>
+        /// <response code="200">Success: A list of all documents</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("Get")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<IEnumerable<DocumentModel>> GetAll()
+        {
+            try
+            {
+                List<DocumentModel> result = _repository.GetAll().ToList();
+                return result.Count == 0 ?
+                    NotFound() :
+                    Ok(result);
+            }
```

Comment:
Hvis i vil gøre det ens alle steder, så kan i formatere det som på linje 37
```suggestion
                return result.Count == 0
                    ? NotFound()
                    : Ok(result);
            }
```